src/x509_crl_generator.ts

```diff
@@ -0,0 +1,123 @@
+import { AsnConvert } from "@peculiar/asn1-schema";
+import * as asn1X509 from "@peculiar/asn1-x509";
+import { container } from "tsyringe";
+import { cryptoProvider } from "./provider";
+import { AlgorithmProvider, diAlgorithmProvider } from "./algorithm";
+import { Extension } from "./extension";
+import { Name } from "./name";
+import { HashedAlgorithm } from "./types";
+import { diAsnSignatureFormatter, IAsnSignatureFormatter } from "./asn_signature_formatter";
+import { X509CrlEntry, X509CRLReason } from "./x509_crl_entry";
+import {  RevokedCertificate, Time } from "@peculiar/asn1-x509";
+import { X509Crl } from "./x509_crl";
+import { X509CertificateCreateParamsName } from "./x509_cert_generator";
+import { PemData } from "./pem_data";
+
+interface X509CrlEntryParamsBase {
+  /**
+   * Hexadecimal serial number
+   */
+  serialNumber: string;
+  revocationDate: Date;
+}
+
+interface X509CrlEntryParams extends X509CrlEntryParamsBase {
+  reason?: X509CRLReason;
+  invalidity?: Date;
+  issuer?: X509CertificateCreateParamsName;
+}
+
+export type X509CrlEntryParamsForCreate = X509CrlEntry[] | X509CrlEntryParams[];
```

Comment:
```diff
- interface X509CrlEntryParamsBase {
-   /**
-    * Hexadecimal serial number
-    */
-   serialNumber: string;
-   revocationDate: Date;
- }
- 
- interface X509CrlEntryParams extends X509CrlEntryParamsBase {
-   reason?: X509CRLReason;
-   invalidity?: Date;
-   issuer?: X509CertificateCreateParamsName;
- }
- 
- export type X509CrlEntryParamsForCreate = X509CrlEntry[] | X509CrlEntryParams[];
+ interface X509CrlEntryParams {
+   /**
+    * Hexadecimal serial number
+    */
+   serialNumber: string;
+   revocationDate?: Date;
+   reason?: X509CRLReason;
+   invalidity?: Date;
+   issuer?: X509CertificateCreateParamsName;
+   extensions?: Extension[];
+ }
```

And don't forget to set default date to `now` for optional `revocationDate`. `Time` doesn't set the default value.
```diff
- revocationDate: new Time(entry.revocationDate),
revocationDate: new Time(entry.revocationDate || new Date()),
```